main.go

```diff
@@ -99,6 +112,7 @@ func main() {
 			return
 		default:
 		}
+		rlimit.Wait(context.Background()) // Waiting a bucket token
```

Comment:
Any idea if this adds any overhead when no limiter is set? Might be safer to have a nil *Limiter when it's not set?